compiler-rt/lib/builtins/CMakeLists.txt

```diff
@@ -906,7 +907,7 @@ else ()
 
       # For RISCV32, we must force enable int128 for compiling long
       # double routines.
-      if(COMPILER_RT_ENABLE_SOFTWARE_INT128 OR "${arch}" STREQUAL "riscv32")
+      if(COMPILER_RT_ENABLE_SOFTWARE_INT128 OR "${arch}" STREQUAL "riscv32" OR "${arch}" STREQUAL "i386")
         list(APPEND BUILTIN_CFLAGS_${arch} -fforce-enable-int128)
```

Comment:
Do we support other compilers? If so enabling this unconditionally for i386 might break them?